src/game_objects/grid.cpp

```diff
@@ -60,8 +53,26 @@ void Grid::setup_listeners() {
         events::build::request_build_event(c);
     });
 
-    events::build::construct_changed_event.connect([&](ConstructType type) {
+    events::build::construct_selected_event.connect([&](ConstructType type) {
         selected = type;
+
+        // Set preview construct accordingly
+        switch(type) {
```

Comment:
This seems to be ugly to maintain. Can we perhaps dispatch the construct instance and polymorphically retrieve the preview?